packages/web3torrent/src/library/web3torrent-lib.ts

```diff
@@ -25,18 +30,37 @@ export * from './types';
 
 export const REQUEST_RATE = 10;
 
+if (process.env.REACT_APP_FAKE_CHANNEL_PROVIDER === 'true') {
+  window.channelProvider = new FakeChannelProvider();
+} else {
+  // TODO: Replace with injection via other means than direct app import
+  // NOTE: This adds `channelProvider` to the `Window` object
+  require('@statechannels/channel-provider');
+}
+
+// TODO: Put inside better place than here where app can handle error case
+window.channelProvider.enable(process.env.REACT_APP_WALLET_URL);
+
 export default class WebTorrentPaidStreamingClient extends WebTorrent {
   allowedPeers: PeersByTorrent;
   pseAccount: string;
   torrents: PaidStreamingTorrent[] = [];
+  channelClient: Web3TorrentChannelClientInterface;
+  amSeeder: boolean = false;
 
   constructor(opts: WebTorrentPaidStreamingClientOptions = {}) {
     super(opts);
+    this.channelClient = new Web3TorrentChannelClient(new ChannelClient(window.channelProvider));
     this.allowedPeers = {};
     this.pseAccount = opts.pseAccount || Math.floor(Math.random() * 99999999999999999).toString();
     log('ACCOUNT ID: ', this.pseAccount);
   }
 
+  async enable() {
+    this.pseAccount = await this.channelClient.getAddress();
```

Comment:
I'm pretty sure we want to use the wallet address for the torrenting account ids, and not to introduce another system.

Reply:
Completely agree on this